pkg/controlsvc/controlsvc.go

```diff
@@ -125,12 +125,14 @@ func (s *Server) AddControlFunc(name string, cType ControlCommandType) error {
 
 // RunControlSession runs the server protocol on the given connection.
 func (s *Server) RunControlSession(conn net.Conn) {
-	logger.Info("Client connected to control service\n")
+	logger.Info("Client connected to control service %s\n", conn.RemoteAddr().String())
```

Comment:
Could that just be 
`logger.Info("Client connected to control service %v\n", conn.RemoteAddr())`